gcloud/bigtable/row.py

```diff
@@ -31,3 +32,111 @@ class Row(object):
     def __init__(self, row_key, table):
         self._row_key = _to_bytes(row_key)
         self._table = table
+
+
+class RowFilter(object):
+    """Basic filter to apply to cells in a row.
+
+    These values can be combined via :class:`RowFilterChain`,
+    :class:`RowFilterUnion` and :class:`ConditionalRowFilter`.
+
+    .. note::
+
+        This class is a do-nothing base class for all row filters.
+    """
+
+    def to_pb(self):
+        """Converts the :class:`RowFilter` to a protobuf.
+
+        :raises: :class:`NotImplementedError <exceptions.NotImplementedError>`
+                 always since a virtual class.
+        """
+        raise NotImplementedError
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+
+class _RegexFilter(RowFilter):
+    """Row filter that uses a regular expression.
+
+    The ``regex`` must be valid RE2 patterns. See Google's
+    `RE2 reference`_ for the accepted syntax.
+
+    .. _RE2 reference: https://github.com/google/re2/wiki/Syntax
+
+    :type regex: bytes or str
+    :param regex: A regular expression (RE2) for some row filter.
+    """
+
+    def __init__(self, regex):
+        self.regex = regex
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        return other.regex == self.regex
+
+    def to_pb(self):
```

Comment:
This re-implementation is a "sacrifice to the `pylint` god" as @tseaver would say.

It wants all virtual methods to be implemented by subclasses. I could also just remove `to_pb` from both bases `RowFilter` and `_RegexFilter`. `NotImplementedError` is not really useful anyhow